### PR TITLE
trace/httptrace: use lib/log

### PR DESCRIPTION
**Pull request by bobheadxi**
Use `lib/log.Logger` in `httptrace.HTTPMiddleware`

## Test plan

main dry run: https://buildkite.com/sourcegraph/sourcegraph/builds/144657

### CODE REVIEW

File: internal/trace/httptrace.go [hunk 1]
```diff
@@ -191,10 +203,10 @@ func HTTPMiddleware(next http.Handler, siteConfig conftypes.SiteConfigQuerier) h
 				routeName = "graphql: unknown"
 			}
 		}
-
-		// route name is only known after the request has been handled
+		logger = logger.With(log.String("route_name", routeName))
```
Comment (keegancsmith) [inline internal/trace/httptrace.go:206]: do any children see this? It looks like there is only one call to logger here, which means if this logger is propogated anywhere this with call is kinda weird? Wouldn't it be more clear to just include the routeName in the only call to logger.Warn?

**Review by keegancsmith — COMMENTED** [review]
do we still panic if log.Init isn't called. Adding it to something like this seems scary then :)

**Comment by bobheadxi** [comment]
> do we still panic if log.Init isn't called. Adding it to something like this seems scary then :)

We only panic in dev to make it obvious something is missing! In prod, it retrieves a no-op logger

File: internal/trace/httptrace.go (hunk 1 above)
Comment (bobheadxi) [reply internal/trace/httptrace.go]: Good point! I've consolidated this all in that one log call

**Comment by bobheadxi** [comment]
Current dependencies on/for this PR:
* main
  * **PR #34594** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/34594" target="_blank">Graphite</a>  (this PR)
    * **PR #34657** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/34657" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/34594?utm_source=stack-comment).

**Review by keegancsmith — APPROVED** [review]

**Pull request merged**